integration/apply-gradient-recommendation.py

```diff
@@ -0,0 +1,114 @@
+# MAGIC %pip install databricks-sdk
+
+# COMMAND ----------
+
+dbutils.library.restartPython()
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## Set Credentials and Environment Variables
+
+# COMMAND ----------
+
+# Collect cluster name
+dbutils.widgets.text("db_workspace_url", "")
+dbutils.widgets.text("job_id", "")
+dbutils.widgets.text("recommendation_id", "")
+
+# COMMAND ----------
+
+import os
+os.environ["DATABRICKS_HOST"] = dbutils.widgets.get("db_workspace_url")
+os.environ["SYNC_RECOMMENDATION_ID"] = dbutils.jobs.taskValues.get(taskKey = "get_sync_recommendation", key = "recommendation_id")
+os.environ["DATABRICKS_JOB_ID"] = dbutils.widgets.get("job_id")
+
+os.environ["DATABRICKS_TOKEN"] = dbutils.secrets.get(scope = "demo_workflow", key = "db_token")
+os.environ["SYNC_API_KEY_ID"] = dbutils.secrets.get(scope = "demo_workflow", key = "sync_access_key_id")
+os.environ["SYNC_API_KEY_SECRET"] = dbutils.secrets.get(scope = "demo_workflow", key = "sync_secret_access_key")
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC
+# MAGIC ## Retrieving Recommendation from Gradient
+
+# COMMAND ----------
+
+import sync
+from sync.api.predictions import *
+
+recommendation = sync.api.predictions.get_prediction(os.getenv("SYNC_RECOMMENDATION_ID"))
+recommended_config = recommendation.result.get('solutions').get('recommended')
+display(recommended_config)
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC
+# MAGIC ## Retrieve Job Cluster Definition from Databricks
+
+# COMMAND ----------
+
+from databricks.sdk import WorkspaceClient
+
+w = WorkspaceClient(
+)
+
+job = w.jobs.get(os.getenv("DATABRICKS_JOB_ID"))
+
+#Current Configuration
+print("Name:" + str(job.settings.job_clusters[0].job_cluster_key or ""))
+print("Driver:" + str(job.settings.job_clusters[0].new_cluster.driver_node_type_id or ""))
+print("Worker:" + str(job.settings.job_clusters[0].new_cluster.node_type_id or ""))
+print("Num Workers:" + str(job.settings.job_clusters[0].new_cluster.num_workers or ""))
+print("Autoscale:" + str(job.settings.job_clusters[0].new_cluster.autoscale or ""))
+print("SparkConf:" + str(job.settings.job_clusters[0].new_cluster.spark_conf or ""))
+
+# COMMAND ----------
+
+from databricks.sdk.service.compute import AutoScale
+
+job.settings.job_clusters[0].new_cluster.driver_node_type_id = recommended_config.get('configuration').get('driver_node_type_id')
+job.settings.job_clusters[0].new_cluster.node_type_id = recommended_config.get('configuration').get('node_type_id')
+if job.settings.job_clusters[0].new_cluster.num_workers is not None:
+    job.settings.job_clusters[0].new_cluster.num_workers = recommended_config.get('configuration').get('num_workers')
+if job.settings.job_clusters[0].new_cluster.autoscale is not None:
+    job.settings.job_clusters[0].new_cluster.autoscale = AutoScale.from_dict(recommended_config.get('configuration').get('autoscale'))
+job.settings.job_clusters[0].new_cluster.spark_conf = recommended_config.get('configuration').get('spark_conf')
```

Comment:
We have a function in the developer interface that similarly applies a prediction to the job: https://github.com/synccomputingcode/syncsparkpy/blob/eaa3039d573e5a50a91d97dbb6bd723373075e96/sync/_databricks.py#L489
```
>>> from sync import awsdatabricks
>>> import orjson
>>> response = awsdatabricks.get_prediction_job(926350091914688, "f8594dc8-0267-45d4-bbab-eabaf257683a", "recommended")                                                              
>>> print(orjson.dumps(response.result, option=orjson.OPT_INDENT_2).decode())
{                                       
  "job_id": 926350091914688,                           
  "run_as_user_name": "scott.romney@synccomputing.com",
  "run_as_owner": true,
  "settings": {                                     
    "name": "scott-load_all_lending_tables",                                    
    "email_notifications": {                                       
      "no_alert_for_skipped_runs": false                        
    },                                                                     
    "webhook_notifications": {                                         
      "on_success": [                            
        {                           
          "id": "b6fa6b0c-bbf6-489c-b4c8-4a06035fac60"
        }                        
      ]      
    },                                
...
```
The above code will update the job cluster whether it's attached to the job or task, and it'll merge values such that spark configurations not in the recommendation won't get overwritten.